pkg/spacebinding/spacebindinglister_test.go

```diff
@@ -235,4 +235,207 @@ func TestNewSpaceBindingLister(t *testing.T) {
+		spaceBMur := NewMasterUserRecord(t, "space-b-user", TargetCluster(test.MemberClusterName), TierName("deactivate90"))
+		spaceCMur := NewMasterUserRecord(t, "space-c-user", TargetCluster(test.MemberClusterName), TierName("deactivate90"))
+		spaceDMur := NewMasterUserRecord(t, "space-d-user", TargetCluster(test.MemberClusterName), TierName("deactivate90"))
+		spaceFMur := NewMasterUserRecord(t, "space-f-user", TargetCluster(test.MemberClusterName), TierName("deactivate90"))
+		// we have a parent space
+		spaceA := spacetest.NewSpace(test.HostOperatorNs, "Space-A",
+			spacetest.WithTierName("advanced"),
+			spacetest.WithSpecTargetCluster(test.MemberClusterName),
+			spacetest.WithDisableInheritance(false),
+			spacetest.WithLabel(toolchainv1alpha1.SpaceCreatorLabelKey, "space-a-user"),
+		)
+		spaceB := spacetest.NewSpace(test.HostOperatorNs, "Space-B",
+			spacetest.WithTierName("advanced"),
+			spacetest.WithSpecTargetCluster(test.MemberClusterName),
+			spacetest.WithSpecParentSpace(spaceA.GetName()),
+			spacetest.WithDisableInheritance(true),
+		)
+		spaceC := spacetest.NewSpace(test.HostOperatorNs, "Space-C",
+			spacetest.WithTierName("advanced"),
+			spacetest.WithSpecTargetCluster(test.MemberClusterName),
+			spacetest.WithSpecParentSpace(spaceB.GetName()),
+			spacetest.WithDisableInheritance(false),
+		)
+		spaceD := spacetest.NewSpace(test.HostOperatorNs, "Space-D",
+			spacetest.WithTierName("advanced"),
+			spacetest.WithSpecTargetCluster(test.MemberClusterName),
+			spacetest.WithSpecParentSpace(spaceC.GetName()),
+			spacetest.WithDisableInheritance(false),
+		)
+		spaceE := spacetest.NewSpace(test.HostOperatorNs, "Space-E",
+			spacetest.WithTierName("advanced"),
+			spacetest.WithSpecTargetCluster(test.MemberClusterName),
+			spacetest.WithSpecParentSpace(spaceD.GetName()),
+			spacetest.WithDisableInheritance(true),
+		)
+		spaceF := spacetest.NewSpace(test.HostOperatorNs, "Space-F",
+			spacetest.WithTierName("advanced"),
+			spacetest.WithSpecTargetCluster(test.MemberClusterName),
+			spacetest.WithSpecParentSpace(spaceE.GetName()),
+			spacetest.WithDisableInheritance(false),
+		)
+		spaces := map[string]*toolchainv1alpha1.Space{
+			spaceA.GetName(): spaceA,
+			spaceB.GetName(): spaceB,
+			spaceC.GetName(): spaceC,
+			spaceD.GetName(): spaceD,
+			spaceE.GetName(): spaceE,
+			spaceF.GetName(): spaceF,
+		}
+		// the getSpaceFunc returns the space based on the given name
+		getSpaceFunc := func(spaceName string) (*toolchainv1alpha1.Space, error) {
+			if space, found := spaces[spaceName]; found {
+				return space, nil
+			}
+			return nil, fmt.Errorf("space not found: %s", spaceName)
+		}
+
+		// listSpaceBindingFunc returns the specific spacebindings for the given space
+		listSpaceBindingFunc := func(spaceName string) ([]toolchainv1alpha1.SpaceBinding, error) {
+			switch spaceName {
+			case spaceA.GetName():
+				fmt.Println("return binding for SpaceA")
```

Comment:
do we need all these `fmt.Println` lines? if we need to log some messages, then let's use `t.Log` instead, but still, I'm not sure if it's needed in this case.

Reply:
removed.